tracer/src/Datadog.Trace.Tools.Analyzers/PublicApiAnalyzer/PublicApiAnalyzer.cs

```diff
@@ -0,0 +1,312 @@
+
+            private static ISymbol? GetOperationSymbol(IOperation operation)
+                => operation switch
+                {
+                    IInvocationOperation iOperation => iOperation.TargetMethod,
+                    IObjectCreationOperation cOperation => cOperation.Constructor,
+                    IFieldReferenceOperation fOperation => IsWithinConditionalOperation(fOperation) ? null : fOperation.Field,
+                    IMemberReferenceOperation mOperation => mOperation.Member,
+                    _ => null,
+                };
+
+            private static IEnumerable<ISymbol?> GetPropertyAccessors(IPropertySymbol property, IOperation operation)
+            {
+                var usageInfo = operation.GetValueUsageInfo(property.ContainingSymbol);
+
+                // not checking/using ValueUsageInfo.Reference related values as property cannot be used as ref or out parameter
+                // not using ValueUsageInfo.Name too, it only use name of the property
+                if (usageInfo == ValueUsageInfo.ReadWrite)
+                {
+                    yield return property.GetMethod;
+                    yield return property.SetMethod;
+                }
+                else if (usageInfo.IsWrittenTo())
+                {
+                    yield return property.SetMethod;
+                }
+                else if (usageInfo.IsReadFrom())
+                {
+                    yield return property.GetMethod;
+                }
+                else
+                {
+                    yield return property;
+                }
+            }
+
+            private static ISymbol? GetEventAccessor(IEventSymbol iEvent, IOperation operation)
+            {
+                if (operation.Parent is IEventAssignmentOperation eventAssignment)
+                {
+                    return eventAssignment.Adds
+                               ? iEvent.AddMethod
+                               : iEvent.RemoveMethod;
+                }
+
+                return iEvent;
+            }
+
+            private static PublicApiStatus CopyAttributes(PublicApiStatus copyAttributes) =>
+                new()
+                {
+                    IsAssemblyAttribute = copyAttributes.IsAssemblyAttribute,
+                    IsPublicApi = copyAttributes.IsPublicApi
+                };
+
+            private static bool IsWithinConditionalOperation(IFieldReferenceOperation pOperation) =>
+                pOperation.ConstantValue.HasValue &&
+                pOperation.Parent is IBinaryOperation bo &&
+                (bo.OperatorKind == BinaryOperatorKind.Equals ||
+                 bo.OperatorKind == BinaryOperatorKind.NotEquals ||
+                 bo.OperatorKind == BinaryOperatorKind.GreaterThan ||
+                 bo.OperatorKind == BinaryOperatorKind.LessThan ||
+                 bo.OperatorKind == BinaryOperatorKind.GreaterThanOrEqual ||
+                 bo.OperatorKind == BinaryOperatorKind.LessThanOrEqual);
```

Comment:
Just for readability, I think you can use
```
bo.OperatorKind is BinaryOperatorKind.Equals or
                   BinaryOperatorKind.NotEquals or
                   BinaryOperatorKind.GreaterThan
                   ...
```

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/3704/commits/1125aca7af68a27c22ebd94969404e4ae8356cea